fhir-profile/src/main/java/org/linuxforhealth/fhir/profile/ProfileSupport.java

```diff
@@ -280,8 +280,15 @@ public static List<Constraint> getConstraints(String url, Class<?> type) {
 
     private static List<Constraint> getConstraints(StructureDefinition profile, Class<?> type) {
         String url = profile.getUrl().getValue();
-        String version = profile.getVersion().getValue();
-        CacheKey key = key(url + "|" + version);
+        CacheKey key;
+        if(profile.getVersion()!=null) {
+            key = key(url + "|" + profile.getVersion().getValue());
+        }
+        else
+        {
+            key = key(url);
+        }
```

Comment:
turning robin's comment into a github suggestion:
```suggestion
        final CacheKey key;
        if(profile.getVersion()!=null) {
            key = key(url + "|" + profile.getVersion().getValue());
        } else {
            key = key(url);
        }
```

Reply:
I also added the profile.getVersion().hasValue() check which covers an extremely rare edge case where the StructureDefinition has a version, but that version has an extension but no value